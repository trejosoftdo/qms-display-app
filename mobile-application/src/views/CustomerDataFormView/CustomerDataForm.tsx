import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Card, TextInput } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import { EMPTY_VALUE } from '../../common/constants';
import { TRANSLATION_CONTINUE_KEY, TRANSLATION_NAME_KEY } from '../../common/translations/translation-keys';
import ActionButton from '../../common/components/ActionButton';
import AppLightTheme from '../../common/theme';

/**
 * CustomerDataForm Props defines the props for the Customer Data Form Component.
 *
 * @property onSubmit - form on submit handler
 */
interface CustomerDataFormProps {
  onSubmit: (name: string) => void;
}

/**
 * A component a customer data form
 *
 * @param {CustomerDataFormProps} props - The props for the Customer Data Form component.
 */
const CustomerDataForm: React.FC<CustomerDataFormProps> = (props: CustomerDataFormProps) => {
  const { t } = useTranslation();
  const [name, setName] = React.useState(EMPTY_VALUE);
  return (
    <Card style={styles.container} mode="outlined">
      <Card.Content>
        <TextInput
          style={styles.input}
          label={t(TRANSLATION_NAME_KEY)}
          value={name}
          onChangeText={text => {
            setName(text);
          }}
          mode="flat"
          left={<TextInput.Icon icon="account" />}
        />
      </Card.Content>
      <Card.Actions style={styles.action}>
        <ActionButton
          icon="page-next"
          message={t(TRANSLATION_CONTINUE_KEY)}
          onPress={() => {
            props.onSubmit(name);
            setName(EMPTY_VALUE);
          }}
        />
      </Card.Actions>
    </Card>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    maxWidth: 500,
    backgroundColor: AppLightTheme.colors.background,
  },
  action: {
    marginBottom: 10,
    marginTop: 10,
    paddingRight: 16,
  },
  input: {
    backgroundColor: AppLightTheme.colors.background,
  }
});


export default CustomerDataForm;