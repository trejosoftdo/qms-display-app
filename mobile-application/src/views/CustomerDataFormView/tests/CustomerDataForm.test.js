import React from 'react';
import { act } from '@testing-library/react-hooks';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import CustomerDataForm from '../CustomerDataForm';


describe('CustomerDataForm component', () => {
  const mockCustomerName = 'mock-customer-name';
  const mockProps = {
    onSubmit: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the component as expected', () => {
    expect(renderShallow(<CustomerDataForm {...mockProps} />)).toMatchSnapshot();
    expect(mockProps.onSubmit).toHaveBeenCalledTimes(0);
  });

  it('changes the language to the selected option', async () => {
    const { getByTestId } = render(<CustomerDataForm {...mockProps} />);
    
    const input = getByTestId('text-input-flat');
    const submitButton = getByTestId('button');
    
    act(() => {
      fireEvent.changeText(input, mockCustomerName);
    });

    await waitFor(() => {
      expect(input.props.value).toEqual(mockCustomerName);
    });
  
    act(() => {
      fireEvent.press(submitButton);
    });

    expect(mockProps.onSubmit).toHaveBeenCalledTimes(1);
    expect(mockProps.onSubmit).toHaveBeenCalledWith(mockCustomerName);
  });
});
