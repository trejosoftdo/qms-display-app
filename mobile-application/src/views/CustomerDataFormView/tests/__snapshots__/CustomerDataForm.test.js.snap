// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDataForm component renders the component as expected 1`] = `
<Card
  mode="outlined"
  style={
    {
      "backgroundColor": "#ffffff",
      "maxWidth": 500,
      "width": "100%",
    }
  }
>
  <Card.Content>
    <ForwardRef
      label="Translated[translation:name]({})"
      left={
        <TextInput.Icon
          forceTextInputFocus={true}
          icon="account"
        />
      }
      mode="flat"
      onChangeText={[Function]}
      style={
        {
          "backgroundColor": "#ffffff",
        }
      }
      value=""
    />
  </Card.Content>
  <Card.Actions
    style={
      {
        "marginBottom": 10,
        "marginTop": 10,
        "paddingRight": 16,
      }
    }
  >
    <ActionButton
      icon="page-next"
      message="Translated[translation:continue]({})"
      onPress={[Function]}
    />
  </Card.Actions>
</Card>
`;
